sympy/polys/matrices/tests/test_lll.py

```diff
@@ -0,0 +1,129 @@
+from sympy.polys.matrices.lll import ddm_lll
+
+normal_test_data = [
+    (
+        DDM(
+            [
+                [ZZ(1), ZZ(0), ZZ(0), ZZ(0), ZZ(-20160)],
+                [ZZ(0), ZZ(1), ZZ(0), ZZ(0), ZZ(33768)],
+                [ZZ(0), ZZ(0), ZZ(1), ZZ(0), ZZ(39578)],
+                [ZZ(0), ZZ(0), ZZ(0), ZZ(1), ZZ(47757)]
+            ],
+            (4, 5),
+            ZZ
+        ),
+        DDM(
+            [
+                [ZZ(10), ZZ(-3), ZZ(-2), ZZ(8), ZZ(-4)],
+                [ZZ(3), ZZ(-9), ZZ(8), ZZ(1), ZZ(-11)],
+                [ZZ(-3), ZZ(13), ZZ(-9), ZZ(-3), ZZ(-9)],
+                [ZZ(-12), ZZ(-7), ZZ(-11), ZZ(9), ZZ(-1)]
+            ],
+            (4, 5),
+            ZZ
+        )
+    ),
+    (
+        DDM(
+            [
+                [ZZ(20), ZZ(52), ZZ(3456)],
+                [ZZ(14), ZZ(31), ZZ(-1)],
+                [ZZ(34), ZZ(-442), ZZ(0)]
+            ],
+            (3, 3),
+            ZZ
+        ),
+        DDM(
+            [
+                [ZZ(14), ZZ(31), ZZ(-1)],
+                [ZZ(188), ZZ(-101), ZZ(-11)],
+                [ZZ(236), ZZ(13), ZZ(3443)]
+            ],
+            (3, 3),
+            ZZ
+        )
+    ),
+    (
+        DDM(
+            [
+                [ZZ(34), ZZ(-1), ZZ(-86), ZZ(12)],
+                [ZZ(-54), ZZ(34), ZZ(55), ZZ(678)],
+                [ZZ(23), ZZ(3498), ZZ(234), ZZ(6783)],
+                [ZZ(87), ZZ(49), ZZ(665), ZZ(11)]
+            ],
+            (4, 4),
+            ZZ
+        ),
+        DDM(
+            [
+                [ZZ(34), ZZ(-1), ZZ(-86), ZZ(12)],
+                [ZZ(291), ZZ(43), ZZ(149), ZZ(83)],
+                [ZZ(-54), ZZ(34), ZZ(55), ZZ(678)],
+                [ZZ(-189), ZZ(3077), ZZ(-184), ZZ(-223)]
+            ],
+            (4, 4),
+            ZZ
+        )
+    )
+
+]
+
+linear_dependent_input = [
+    DDM(
+        [
+            [ZZ(0), ZZ(-1), ZZ(-2), ZZ(-3)],
+            [ZZ(1), ZZ(0), ZZ(-1), ZZ(-2)],
+            [ZZ(2), ZZ(1), ZZ(0), ZZ(-1)],
+            [ZZ(3), ZZ(2), ZZ(1), ZZ(0)]
+        ],
+        (4, 4),
+        ZZ
+    ),
+    DDM(
+        [
+            [ZZ(1), ZZ(0), ZZ(0), ZZ(1)],
+            [ZZ(0), ZZ(1), ZZ(0), ZZ(1)],
+            [ZZ(0), ZZ(0), ZZ(1), ZZ(1)],
+            [ZZ(1), ZZ(2), ZZ(3), ZZ(6)]
+        ],
+        (4, 4),
+        ZZ
+    ),
+    DDM(
+        [
+            [ZZ(3), ZZ(-5), ZZ(1)],
+            [ZZ(4), ZZ(6), ZZ(0)],
+            [ZZ(10), ZZ(-4), ZZ(2)]
+        ],
+        (3, 3),
+        ZZ
+    )
+]
+
+
+def test_lll():
+    for basis, reduced in normal_test_data:
+        assert ddm_lll(basis, delta=QQ(5, 6)) == reduced
+        assert basis.lll(delta=QQ(5, 6)) == reduced
+        assert DomainMatrix.from_rep(basis).lll(delta=Rational(5, 6)).rep == reduced
+
+
+def test_lll_linear_dependent():
+    for not_basis in linear_dependent_input:
+        try:
+            ddm_lll(not_basis)
+        except DMRankError:
+            pass
```

Comment:
For this we use `raises`:
```
from sympy.testing.pytest import raises

def test_stuff():
   raises(DMRankError, lambda: ddm_lll(not_basis)
```
This way we are asserting that the exception does get raised (the test would fail if the exception was not raised).